gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineExactDependencies.java

```diff
@@ -99,11 +101,20 @@ private static void configureSourceSet(
                             compile.toString(), implementation.toString()));
                     conf.setVisible(false);
                     conf.setCanBeConsumed(false);
-                    // Important! this ensures we resolve 'compile' variants rather than 'runtime'
-                    // This is the same attribute that's being set on compileClasspath
-                    conf.getAttributes()
-                            .attribute(
-                                    Usage.USAGE_ATTRIBUTE, project.getObjects().named(Usage.class, Usage.JAVA_API));
+
+                    conf.attributes(attributes -> {
+                        // This ensures we resolve 'compile' variants rather than 'runtime'
+                        // This is the same attribute that's being set on compileClasspath
+                        attributes.attribute(
+                                Usage.USAGE_ATTRIBUTE, project.getObjects().named(Usage.class, Usage.JAVA_API));
+                        // Ensure we resolve the classes directory for local projects where possible, rather than the
+                        // 'jar' file. We can only do this on Gradle 5.6+, otherwise do nothing.
+                        if (GradleVersion.current().compareTo(GradleVersion.version("5.6")) >= 0) {
+                            attributes.attribute(
+                                    LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE,
+                                    project.getObjects().named(LibraryElements.class, LibraryElements.CLASSES));
+                        }
+                    });
```

Comment:
This is mostly an optimisation so we don't force the production of jars since we can use the classes dirs.
Gradle 6.4 has [made it so that](https://github.com/gradle/gradle/blob/42f7c3d0c3066b7b38bd0726760d4881e86fd19f/subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JvmPluginsHelper.java#L250-L255) it sometimes forces jars when resolving e.g. `compileClasspath` in order to be able to confidently extract the `Automatic-Module-Name` if there is one set in the manifest.
However I think we can always just use the classes dir for this case, as it won't change the behaviour.